### PR TITLE
refactor(react): use reducer

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Need a better state management for react SDK, use `useReducer` to manage login flows.

## Testing

reuse `use-logto.spec.tsx`

### CODE REVIEW

File: packages/react/src/auth-state.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export interface AuthState {
+  isLoading: boolean;
+  initialized: boolean;
```
Comment (IceHe) [inline packages/react/src/auth-state.ts:3]: why not startWith `is` like `isLoading` and `isAuthenticated`?

**Comment by gao-sun** [comment]
how about `zustand`? it's tiny and has been tested while I was working for sequoia

**Comment by wangsijie** [comment]
> how about `zustand`? it's tiny and has been tested while I was working for sequoia

native `useReducer` is enough for now (and for the future, oidc process is fixed), can keep SDK bundle small

File: packages/react/src/auth-state.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+export interface AuthState {
```
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/react/src/auth-state.ts:1]: not a big deal leave up to you

[the remark this answers is not in the document]
Comment (wangsijie) [reply packages/react/src/auth-state.ts:1]: keep it for now, we may add sth later, such as `user`

**Review by simeng-li — APPROVED** [review]
LGTM

**Pull request merged**